src/library_pthread.js

```diff
@@ -376,6 +377,14 @@ var LibraryPThread = {
 
       worker.onerror = function(e) {
         err('pthread sent an error! ' + e.filename + ':' + e.lineno + ': ' + e.message);
+#if PROXY_TO_PTHREAD
+        // If the worker running the main thread generates an error, we want
```

Comment:
This doesn't seem to check that it's the worker running the main thread - it seems like any worker hitting an exception will be handled this way?

Reply:
Right.. i was worried about that.  Do message from all worker come back here?  I guess so.   TBH I'm not sure we shouldn't behave in the same way.   In the native world a crash in any thread bring the whole program down.   Users can still catch this exception, but I think we should at least throw it so that by default the process ends.

What do you think?

Reply:
Yes, this is where we get messages from all workers, not just the main() one.

If there's an easy way to check if it's main(), lgtm with that. Otherwise I guess we can discuss in #13000

Reply:
OK .. I added a way to detect the proxy main.